packages/compass-data-modeling/src/components/saved-diagrams-list.tsx

```diff
@@ -96,105 +133,107 @@ export const SavedDiagramsList: React.FunctionComponent<{
 }) => {
   const { items, status } = useDataModelSavedItems();
 
+  const [filteredItems, setFilteredItems] = useState(items);
+  const [sortControls, sortState] = useSortControls(sortBy);
+  const sortedItems = useSortedItems(filteredItems, sortState);
+
+  useEffect(() => {
+    setFilteredItems(items);
+  }, [items]);
+
+  const onSearchItems = useCallback(
+    (search: string) => {
+      try {
+        const regex = new RegExp(search, 'i');
```

Comment:
This is an actual state you need to keep around

Reply:
I think i am not able to follow this, but what's the benefit of storing this (`regex`) in the state? I am already storing `search` value in the state, but in toolbar component.

Reply:
I ended up moving this to the context and now, as you suggested, filtering data within this component in [38dbebf](https://github.com/mongodb-js/compass/pull/6931/commits/38dbebf630f8f83ab1855b2a3b4b1c158110565c)